acceptance/lib/puppet/acceptance/service_utils.rb

```diff
@@ -14,6 +14,57 @@ def supports_systemd? (host)
         suitable = on(host, "#{ruby} -e \"require 'puppet'; puts Puppet::Type.type(:service).provider(:systemd).suitable?\"" ).stdout.chomp
         suitable == "true" ? true : false
       end
+
+      # Alter the state of a service using Puppet.
+      # @param host [String] hostname.
+      # @param service [String] name of the service.
+      # @param property [String] name of the attribute to be changed.
+      # @param value [String] value which the property should be set.
+      # @return None
+      def ensure_service_on_host(host, service, property, value)
+        manifest = %Q{
+          service { '#{service}':
+            #{property} => '#{value}'
+          }
+        }
+        # the process of creating the service will also start it
+        # to avoid a flickering test from the race condition, this test will ensure
+        # that the exit code is either
+        #   2 => something changed, or
+        #   0 => no change needed
+        on host, puppet_apply(['--detailed-exitcodes', '--verbose']),
+          {:stdin => manifest, :acceptable_exit_codes => [0, 2]}
+        # ensure idempotency
+        on host, puppet_apply(['--detailed-exitcodes', '--verbose']),
```

Comment:
Unfortunately I've been bitten by cases where `puppet resource` behaves differently than `apply`, e.g. 1e8a2cd86bafb4c73d699cea67e9605e78ac0d0a, so I'm :+1: for testing both.